### DISCUSSION TITLE
Detecting existing user sign up with Auth UI

### DISCUSSION THREAD

**Discussion by ozzy1136** [supabase/supabase#14166] [Questions] [answered]
As the title states, I want to know if there's any obvious way to get a warning message if someone is trying to sign up with an existing user email. Currently, I have email confirmation enabled for the project and if an existing user email is used to sign up then the warning message says "check your email for confirmation" but no email is sent.

I'm working on my first project with Supabase and Auth UI (React) and I wasn't sure if this is a bug, so I decided to post here. Any help is appreciated!

**Comment by GaryAustin1** [comment]
There is no way to know.  Supabase considers it a security leak to disclose if email exists.
Some users have created their own postgres function to check for email exists using an rpc call.

Long thread of several... https://github.com/orgs/supabase/discussions/1282

**Reply by ozzy1136** [reply, marked as answer]
That makes perfect sense, and thanks for the link to the discussion.
I had only searched in the repo issues expecting more questions from newbs like me. Thanks!
